x-pack/filebeat/tests/system/test_http_endpoint.py

```diff
@@ -138,6 +138,52 @@ def test_http_endpoint_wrong_auth_value(self):
         assert r.status_code == 401
         assert r.text == '{"message": "Incorrect username or password"}'
 
+    def test_http_endpoint_wrong_auth_header(self):
+        """
+        Test http_endpoint input with wrong auth header and secret.
+        """
+        options = """
+  secret.header: Authorization
+  secret.value: 123password
+"""
+        self.get_config(options)
+        filebeat = self.start_beat()
+        self.wait_until(lambda: self.log_contains("Starting HTTP server on {}:{}".format(self.host, self.port)))
+
+        message = "somerandommessage"
+        payload = {self.prefix: message}
+        headers = {"Content-Type": "application/json", "Authorization": "password123"}
+        r = requests.post(self.url, headers=headers, data=json.dumps(payload))
+
+        filebeat.check_kill_and_wait()
+
+        assert r.status_code == 401
+        assert r.text == '{"message": "Incorrect header or header secret"}'
+
+    def test_http_endpoint_correct_auth_header(self):
+        """
+        Test http_endpoint input with correct auth header and secret.
+        """
+        options = """
+  secret.header: Authorization
+  secret.value: 123password
+"""
+        self.get_config(options)
+        filebeat = self.start_beat()
+        self.wait_until(lambda: self.log_contains("Starting HTTP server on {}:{}".format(self.host, self.port)))
+
+        message = "somerandommessage"
+        payload = {self.prefix: message}
+        headers = {"Content-Type": "application/json", "Authorization": "123password"}
+        r = requests.post(self.url, headers=headers, data=json.dumps(payload))
+
+        filebeat.check_kill_and_wait()
+        output = self.read_output()
+
+        assert r.text == '{"message": "success"}'
+        assert output[0]["input.type"] == "http_endpoint"
+        assert output[0]["json.{}".format(self.prefix)] == message
```

Comment:
Python based tests tend to be slow and possibly flaky. It would be nice if we could introduce go based unit tests instead.

Reply:
Talked with @urso about this, will remake these into go based tests before 7.10, but want to do it in another PR as there is already nosetests for these and need the input functionality for the zoom module to work first.